nifi-nar-bundles/nifi-jslt-bundle/nifi-jslt-processors/src/main/java/org/apache/nifi/processors/jslt/JSLTTransformJSON.java

```diff
@@ -0,0 +1,269 @@
+        if (!validationContext.getProperty(JSLT_TRANSFORM).isExpressionLanguagePresent()) {
+            final String transform = validationContext.getProperty(JSLT_TRANSFORM).getValue();
+            try {
+                Parser.compileString(transform);
+            } catch (JsltException je) {
+                results.add(new ValidationResult.Builder().subject(JSLT_TRANSFORM.getDisplayName()).valid(false).explanation("error in transform: " + je.getMessage()).build());
+            }
+        } else {
+            // Expression Language is present, we won't know if the transform is valid until the EL is evaluated
+            results.add(new ValidationResult.Builder().subject(JSLT_TRANSFORM.getDisplayName()).valid(true).build());
+        }
+        return results;
+
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        int maxTransformsToCache = context.getProperty(TRANSFORM_CACHE_SIZE).asInteger();
+        transformCache = Caffeine.newBuilder()
+                .maximumSize(maxTransformsToCache)
+                .build();
+        // Precompile the transform if it hasn't been done already (and if there is no Expression Language present)
+        if (!context.getProperty(JSLT_TRANSFORM).isExpressionLanguagePresent()) {
+            final String transform = context.getProperty(JSLT_TRANSFORM).getValue();
+            try {
+                transformCache.put(transform, Parser.compileString(transform));
+            } catch (JsltException je) {
+                throw new ProcessException("Error compiling JSLT transform: " + je.getMessage(), je);
+            }
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile original = session.get();
+        if (original == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+        final StopWatch stopWatch = new StopWatch(true);
+
+        JsonNode firstJsonNode;
+        try (final InputStream in = session.read(original)) {
+            firstJsonNode = readJson(in);
+        } catch (final Exception e) {
+            logger.error("Failed to transform {}; routing to failure", new Object[]{original}, e);
+            session.transfer(original, REL_FAILURE);
+            return;
+        }
+
+        final String jsonString;
+        final String transform;
+        try {
+            transform = context.getProperty(JSLT_TRANSFORM).evaluateAttributeExpressions(original).getValue();
+            Expression jsltExpression = transformCache.get(transform, currString -> Parser.compileString(transform));
+
+            final JsonNode transformedJson = jsltExpression.apply(firstJsonNode);
+            if (transformedJson == null) {
+                jsonString = "";
+                logger.info("JSLT transform resulted in no data!");
```

Comment:
Log message should not include an exclamation mark. It is probably better to remove this line. Should this throw an exception, or at least log a warning?